docs/modules/ROOT/pages/references/parameters.adoc

```diff
@@ -0,0 +1,163 @@
+= Parameters
+
+The parent key for all of the following parameters is `openshift4_nodes`.
+
+[CAUTION]
+====
+This component relies on deep merge of values from several parameters and hierarchy layers.
+This works pretty strait forward for scalar values and dictionaries.
+Values of arrays will be appended to each other.
+There is no way to override values of set on lover precedence location.
+====
+
+== `availabilityZones`
```

Comment:
I'd propose to move this to `parameters.cloud.availabilityZones`, so it's together with `cloud.provider` and `cloud.region`.

Reply:
As discussed: we will keep this parameter in the component scope. Assignments from a global scope can be made within the hierarchy. This reduces the amount of implicit dependencies and has some more advantages (e.g. facts have highest precedence but when having them assigned to a component parameter, they can be overwritten again if need be).